### PR TITLE
feat(api-client): add the button to API client

### PR DESCRIPTION
**Pull request by amritk**
Integration isn't wired up in the client but should be trivial. Also need to decide when to show it

### CODE REVIEW

**Review by marclave — APPROVED** [review]

File: packages/oas-utils/src/entities/spec/collection.ts [hunk 1]
```diff
@@ -41,6 +41,34 @@ const oasCollectionSchema = z.object({
...
+  'litestar',
+  'nestjs',
+  'nextjs',
+  'nitro',
+  'nuxt',
+  'platformatic',
+  'react',
+  'rust',
+  'vue',
+])
```
Comment (hanspagel) [inline packages/oas-utils/src/entities/spec/collection.ts:70]: `null` is valid, too

super annoying to keep those two lists in sync! should we make this just `string | null`? the import modal should gracefully handle other values and fall back to the openapi logo

File: packages/oas-utils/vitest.config.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+import viteConfig from './vite.config'
+
+export default mergeConfig(
+  viteConfig,
+  defineConfig({
+    test: {
+      environment: 'jsdom',
+      root: fileURLToPath(new URL('./', import.meta.url)),
+    },
```
Comment (hanspagel) [inline packages/oas-utils/vitest.config.ts:12]: we can just add those lines to the vite.config.ts, isn’t that easier/cleaner?

File: packages/api-client/src/views/Request/RequestSubpageHeader.vue [hunk 3]
```diff
@@ -33,6 +37,11 @@ defineEmits<{
         target="_blank">
         Powered by Scalar.com
       </a>
+      <!-- TODO: choose when to display it -->
```
Comment (hanspagel) [inline packages/api-client/src/views/Request/RequestSubpageHeader.vue:40]: let’s add `if activeCollection?.documentUrl`

File: packages/api-client/src/components/OpenApiClientButton.vue [hunk 4]
```diff
@@ -1,43 +1,33 @@
...
+  if (integration) link.searchParams.set('integration', integration)
 
   return link.toString()
 })
 </script>
 
 <template>
   <a
-    v-if="getOpenApiDocumentUrlSymbol?.()"
+    v-if="url"
```
Comment (hanspagel) [inline packages/api-client/src/components/OpenApiClientButton.vue:30]: this should maybe be `v-if="href"`, eh?

**Review by hanspagel — CHANGES_REQUESTED** [review]
Cool, just some nit picks

File: packages/oas-utils/src/entities/spec/collection.ts (hunk 1 above)
Comment (amritk) [reply packages/oas-utils/src/entities/spec/collection.ts:70]: yea thats fine

File: packages/oas-utils/vitest.config.ts (hunk 2 above)
Comment (amritk) [reply packages/oas-utils/vitest.config.ts:12]: sure, just copied this out of one of the other repos

File: packages/api-client/src/components/OpenApiClientButton.vue (hunk 4 above)
Comment (amritk) [reply packages/api-client/src/components/OpenApiClientButton.vue:30]: aha probably!

**Comment by amritk** [comment]
There's some rebase shennanigans going on here so I will make a new PR with hans' changes

**Comment by amritk** [comment]
closing in favour of https://github.com/scalar/scalar/pull/3653

**Pull request closed without merging**